swap_meet/vendor.py

```diff
@@ -1,2 +1,77 @@
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item_to_remove):
+        if item_to_remove not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item_to_remove)
+        return item_to_remove
+
+    def get_by_category(self, category):
+        items_by_category = []
+
+        for item in self.inventory:
+            if category == item.category:
+                items_by_category.append(item)
+        return items_by_category
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        vendor.inventory.append(my_item)
+        return True 
+
+    def swap_first_item(self, vendor):
+        if not self.inventory or not vendor.inventory:
+            return False
+
+        self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
+        return True
```

Comment:
You've got a guard clause on lines 37-38 that returns False if inventory is empty and there's no first item to swap. However, if there are first items to swap you reuse your swap_items method -- which returns a boolean. So instead of returning True, you can return the value of calling swap_items:
```python
return self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
```

Also, a quick note about time complexity - It is great to reuse methods we've already written, however swap_items has linear time complexity because it invokes the remove() method.

Can you think of how you could write a custom swapping implementation to bring the time complexity to O(1) ? Recall that Python has syntax for reassigning values like `a, b = b, a`